### PR TITLE
nf-core v2.14 update

### LINKED ISSUE #70: Implement nf-validation
### Description of feature

Implement nf-validation for parameters and sample sheets

### PR DESCRIPTION
**Pull request by tkchafin**
Solves #70 

(1) [nf-core v2.13 migration](https://nf-co.re/blog/2024/tools-2_13): 
- Removal of `lib/` directory in favour of new boilerplate sub workflows
- Switch to `nf_validation` for sample sheet checking + removal of local module for sample sheet checking -> validated sample sheet is now passed directly to `input_check.nf` sub workflow, which just runs `SAMTOOLS_FLAGSTAT` and formats channels for alignment sub workflows.
- Moved start-up and completion steps to new sub workflows `PIPELINE_INITIALISATION` and `PIPELINE_COMPLETION` defined in `subworkflows/local/utils_nfcore_readmapping_pipeline` 

(2) nf-core v2.14 migration
- A number of changes proposed to the `.github/workflows` were rejected, with exception of: (a) added new `download_pipelines.yml`; (b) update nf-core setup version in `ci.yml` 
- Minor changes to a number of config files 
- Full template update (pushed to `TEMPLATE` branch https://github.com/sanger-tol/readmapping/pull/98) 

(3) Other
- nf-core modules and sub workflow updates 
- Update Samtools v1.17 -> v1.20 in local modules 

NOTES: 
1. The `INPUT_CHECK` sub workflow now only has a single module step, so could probably be removed and just call the module directly in `readmapping.nf` 
2. There is a redundant "monochromeLogs" parameter in `nextflow.config`... This is needed by `nf-validation`. This is fixed in [`nf-schema`](https://github.com/nextflow-io/nf-schema/issues/14), however `nf-schema` [can't be used alongside `nf-validation`](https://github.com/nextflow-io/nf-schema/issues/28#issuecomment-2074790184), which is still required by `subworkflows/nf-core/utils_nfvalidation_plugin` so for now I stick to `nf-validation:1.13`

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by tkchafin** [comment]
Still need to fix linting issues, not ready for review

**Comment by tkchafin** [comment]
ready for review

File: nextflow_schema.json
Comment (BethYates) [inline nextflow_schema.json:1]: I think the changes on line 36 and 233 should be reverted?

Comment (tkchafin) [reply nextflow_schema.json:1]: Oops, thought I had reverted that. Just fixed

**Comment by BethYates** [comment]
I'm not sure if this is intended or not but when you run the pipeline using the test profile it now asks for an --outdir param because the default value in modules.config has changed from "./results" to null. This makes sense as the docs say this is a required parameter but do we want to maybe set a default value for  --outdir in the test and test_full profiles?

**Review by BethYates — APPROVED** [review]

**Pull request merged**